### PR TITLE
refactor: Add telemetry for RBAC (no-changelog)

### PR DESCRIPTION
**Pull request by ivov**
https://linear.app/n8n/issue/PAY-1142

### CODE REVIEW

File: packages/cli/src/services/user.service.ts [hunk 1]
```diff
@@ -307,4 +308,22 @@ export class UserService {
 
 		return { usersInvited, usersCreated: toCreateUsers.map(({ email }) => email) };
 	}
+
+	/**
+	 * Counts the number of users in each role, e.g. `{ admin: 2, member: 6, owner: 1 }`
+	 */
+	async countUsersByRole() {
+		const result: Array<{ role_name: string; count: number }> = await Container.get(RoleRepository)
```
Comment (krynble) [inline packages/cli/src/services/user.service.ts:316]: Should we add tests for this method?

**Review by krynble — DISMISSED** [review]
I guess this could be noted with `(no-changelog)` as it's not really important for users.

File: packages/cli/src/services/user.service.ts (hunk 1 above)
Comment (ivov) [reply packages/cli/src/services/user.service.ts]: Done!

File: packages/cli/src/services/role.service.ts [hunk 2]
```diff
@@ -100,4 +100,22 @@ export class RoleService {
...
+			.select('role.name')
+			.addSelect('COUNT(user.id)', 'count')
+			.innerJoin('user', 'user', 'role.id = user.globalRoleId')
+			.groupBy('role.name')
+			.getRawMany();
+
+		return result.reduce<Record<string, number>>((acc, item) => {
+			acc[item.role_name] = item.count;
+			return acc;
+		}, {});
```
Comment (netroy) [inline packages/cli/src/services/role.service.ts:119]: any code generating a query should be move to repository classes. That's the whole purpose of those classes.

File: packages/editor-ui/src/components/InviteUsersModal.vue [hunk 3]
```diff
@@ -333,7 +333,7 @@ export default defineComponent({
 			}
 		},
 		goToUpgradeAdvancedPermissions() {
-			void this.uiStore.goToUpgrade('settings-users', 'upgrade-advanced-permissions');
+			void this.uiStore.goToUpgrade('advanced-permissions', 'upgrade-advanced-permissions');
```
Comment (netroy) [inline packages/editor-ui/src/components/InviteUsersModal.vue:336]: this looks like it fixes an unrelated bug. is there a ticket for that?

**Review by netroy — CHANGES_REQUESTED** [review]

File: packages/editor-ui/src/components/InviteUsersModal.vue (hunk 3 above)
Comment (ivov) [reply packages/editor-ui/src/components/InviteUsersModal.vue:336]: It's related to this. Should this be a separate PR?

Capture 2023-12-19 at 16 25 35@2x

**Comment by netroy** [comment]
LGTM, but there is a conflict.

**Review by netroy — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.22.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.22.0)

**Pull request merged**